hedera-node/docs/design/services/token-service/airdrops/airdrop-transaction.md

```diff
@@ -0,0 +1,206 @@
+          - an existing account, which was not deleted or expired
+          - a non-existing account, pointed by an alias of `evm_address` or public `ECDSA key`
+        - Check that the token being airdropped, does not contain a custom fee, which needs to be covered by the recipient. Such transactions should be reverted and rejected.
+        - The business logic for sending pending airdrops
+            - Check the association status and the existence of the recipient:
+
+              - It is an existing one and has `max_auto_associations` set to -1 and is not associated to the token →
+                the token is auto associated and is directly transferred to the recipient and a `TokenTransferList` is added to the `TransactionRecord`, as well as a `TokenAssociation`
+
+              - It is an existing one and has `max_auto_associations`, which is a positive number and there are free auto association slots and the recipient is not associated to the token →
+                the token is auto associated and is directly transferred to the recipient and a `TokenTransferList` is added to the `TransactionRecord`, as well as a `TokenAssociation`
+
+              - It is an existing one and has `max_auto_associations`, which is a positive number but there are no free auto association slots and the recipient is not associated to the token →
+                the token transfer is interpreted as a pending and a `TokenPendingAirdrop` entry is added to the `TransactionRecord`
+
+              - It is an existing one and has `max_auto_associations` set to 0 and the recipient is not associated to the token →
+                the token transfer is interpreted as a pending and a `TokenPendingAirdrop` entry is added to the `TransactionRecord`
+
+              - It is an existing one and is associated to the token →
+                the airdrop is handled as a regular crypto transfer and a `TokenTransferList` is added to the `TransactionRecord`
+
+              - It’s not existing and the AccountID is a public `ECDSA` key →
+                the airdrop would first auto-create a new account with `maxAutoAssociations` field of -1 and directly transfer the token being airdropped to it. This would include a `TokenTransferList` and the alias of the newly created account in the `TransactionRecord`
+
+              - It’s not existing and the AccountID is an `evm_address` →
+                the airdrop would first lazily-create a new account with `maxAutoAssociations` field of -1 and directly transfer the token being airdropped to it. This would include a `TokenTransferList` and the alias of the newly created account in the `TransactionRecord`
+    - Fee calculation logic
+        - assess all required fees for the airdrop from the payer of the transaction. This includes:
+            - airdrop fee (assessed always)
+            - first auto-renew fee for the token (assessed always)
+            - custom token fees covered by the sender (assessed always if present)
```

Comment:
Do we store these custom fees somewhere? When are the custom fees charged ?

Reply:
I propose not to store them. They would be charged in the `TokenAirdrop.calculateFees` during the sending of airdrop